src/actions.rs

```diff
@@ -0,0 +1,497 @@
+    Up {
+        /// The mouse button index. Button values are as follows:
+        /// - Left = 0
+        /// - Middle = 1
+        /// - Right = 2
+        button: u64,
+    },
+    /// Pointer move action.
+    ///
+    /// The x and y offsets are relative to the origin.
+    Move {
+        /// The move duration.
+        duration: Option<Duration>,
+        /// The origin that the `x` and `y` coordinates are relative to.
+        origin: PointerOrigin,
+        /// `x` offset, in pixels.
+        x: i64,
+        /// `y` offset, in pixels.
+        y: i64,
+    },
+    /// Pointer cancel action. Used to cancel the current pointer action.
+    Cancel,
+}
+
+/// The pointer origin to use for the relative x,y offset.
+#[derive(Debug, Clone)]
+#[non_exhaustive]
+pub enum PointerOrigin {
+    /// Coordinates are relative to the top-left corner of the browser window.
+    Viewport,
+    /// Coordinates are relative to the pointer's current position.
+    Pointer,
+    /// Coordinates are relative to the specified element's center position.
+    WebElement(Element),
+}
+
+impl PointerOrigin {
+    fn into_wd_pointerorigin(self) -> WDActions::PointerOrigin {
+        match self {
+            PointerOrigin::Viewport => WDActions::PointerOrigin::Viewport,
+            PointerOrigin::Pointer => WDActions::PointerOrigin::Pointer,
+            PointerOrigin::WebElement(e) => WDActions::PointerOrigin::Element(
+                webdriver::common::WebElement(e.element_id().into()),
+            ),
+        }
+    }
+}
+
+impl PointerAction {
+    fn into_item(self) -> WDActions::PointerActionItem {
+        match self {
+            PointerAction::Pause { duration } => WDActions::PointerActionItem::General(
+                WDActions::GeneralAction::Pause(WDActions::PauseAction {
+                    duration: Some(duration.as_millis() as u64),
+                }),
+            ),
+            PointerAction::Down { button } => WDActions::PointerActionItem::Pointer(
+                WDActions::PointerAction::Down(WDActions::PointerDownAction { button }),
+            ),
+            PointerAction::Up { button } => WDActions::PointerActionItem::Pointer(
+                WDActions::PointerAction::Up(WDActions::PointerUpAction { button }),
+            ),
+            PointerAction::Move {
+                duration,
+                origin,
+                x,
+                y,
+            } => WDActions::PointerActionItem::Pointer(WDActions::PointerAction::Move(
+                WDActions::PointerMoveAction {
+                    duration: duration.map(|x| x.as_millis() as u64),
+                    origin: origin.into_wd_pointerorigin(),
+                    x: Some(x),
+                    y: Some(y),
+                },
+            )),
+            PointerAction::Cancel => {
+                WDActions::PointerActionItem::Pointer(WDActions::PointerAction::Cancel)
+            }
+        }
+    }
+}
+
+/// A sequence containing `Null` actions.
```

Comment:
If you write:
```
[`NullAction`]
```
instead, it'll become a link in the documentation. You can, I think, even write:
```rust
/// A sequence containing [`Null` actions](NullAction).

Reply:
We should probably do this on the other `*Actions` structs.